src/ape/types/__init__.py

```diff
@@ -21,6 +21,14 @@
 as well as a block number or hash (HexBytes).
 """
 
+SnapshotID = Union[str, int, bytes]
+"""
+An ID representing a point in time on a blockchain, as used in the
+:meth:`~ape.managers.chain.ChainManager.snapshot` and
+:meth:`~ape.managers.chain.ChainManager.snapshot` methods. Can be a ``str``, ``int``, or ``bytes``.
+Providers will expect and handle snapshot IDs differently.
```

Comment:
Add "There shouldn't be a need to change providers when using this feature, so there should not be confusion over this type in practical use cases"